trees/redblacktree/redblacktree.go

```diff
@@ -31,7 +32,7 @@ const (
 type Tree struct {
 	Root       *Node
 	size       int
-	Comparator utils.Comparator
+	comparator utils.Comparator
```

Comment:
Maybe exposing the Comparator is easier than hiding it an then providing a getter for it? Based on my Java experience, I found getters/setters to be evil.